changelogs/unreleased/fix-tx-rv-and-conflicted-states.md

```diff
@@ -0,0 +1,11 @@
+## bugfix/box
+
+* Made read-view transactions become conflicted on attempt to perform DML
+  statements immediately: previously this was detected only on transaction
+  preparation stage, i.e., when calling `box.commit` (gh-7240).
```

Comment:
This looks like a breaking change feature. I think we should split the change log in two:
- `feature/core` (we don't have `box` [changelog section](https://github.com/tarantool/tarantool/blob/ede831d317c3a5ba02f202a7daf387742ddbfeb6/tools/gen-release-notes#L15-L17)) marked as "breaking change" for the changed behavior of CRUD operations in aborted transactions.
- `bugfix/memtx` for the invalid result fix.

Reply:
Fixed: split change log into two as you suggested.